types/write-file-atomic/index.d.ts

```diff
@@ -2,25 +2,29 @@
 // Project: https://github.com/npm/write-file-atomic
 // Definitions by: BendingBender <https://github.com/BendingBender>
 //                 Jay Rylan <https://github.com/jayrylan>
+//                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
 export = writeFile;
 
-declare function writeFile(filename: string, data: string | Buffer, options: writeFile.Options, callback: (error?: Error) => void): void;
+declare function writeFile(filename: string, data: string | Buffer, options: writeFile.Options | BufferEncoding, callback: (error?: Error) => void): void;
 declare function writeFile(filename: string, data: string | Buffer, callback: (error?: Error) => void): void;
-declare function writeFile(filename: string, data: string | Buffer, options?: writeFile.Options): Promise<void>;
+declare function writeFile(filename: string, data: string | Buffer, options?: writeFile.Options | BufferEncoding): Promise<void>;
 
 declare namespace writeFile {
-    function sync(filename: string, data: string | Buffer, options?: Options): void;
+    function sync(filename: string, data: string | Buffer, options?: Options | BufferEncoding): void;
 
     interface Options {
         chown?: {
             uid: number;
             gid: number;
         };
-        encoding?: string;
+        /**
+         * @default 'utf8'
+         */
+        encoding?: BufferEncoding | '';
```

Comment:
Just to confirm, `''` is permitted in the options, but not it you specify the encoding directly as an argument?

Reply:
That's not in the package source code, it was me adding this only to be compliant with existing contract from tests (for DT package):
```ts
writeFileAtomic('message.txt', 'Hello Node', {encoding: ''}, err => {});
```
Personally I'd rather remove that contract (and modify options as per your comment).
Thoughs?

Reply:
Generally speaking, I try to write types that reflect the way APIs are intended to be consumed and not to reflect every way they _could_ be called.  If `''` is just another falsey value (i.e. the same as `undefined`), then I would lean towards removing it.  If it's idiomatic for the library (e.g. it's in a sample usage or you find many callers on GH passing `''`), then it seems worth including.

Reply:
amended with 9e24a860c8
Thanks!